superset/daos/key_value.py

```diff
@@ -0,0 +1,195 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import Any
+from uuid import UUID
+
+from sqlalchemy import and_
+
+from superset import db
+from superset.daos.base import BaseDAO
+from superset.key_value.exceptions import (
+    KeyValueCreateFailedError,
+    KeyValueUpdateFailedError,
+)
+from superset.key_value.models import KeyValueEntry
+from superset.key_value.types import KeyValueCodec, KeyValueResource
+from superset.key_value.utils import get_filter
+from superset.utils.core import get_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class KeyValueDAO(BaseDAO[KeyValueEntry]):
```

Comment:
I think the current behavior (though wrong) is for DAOs to commit. Should you mimic that same logic here? I can update the logic in https://github.com/apache/superset/pull/24969 (if it ever lands).

Reply:
Sounds good, I'll update accordingly.

Reply:
@john-bodley Uhh, I started doing this, and updating the code with the old intermediate commit logic is feeling super wrong now that I've gotten used to the new way of doing it 🤮  I'm wondering if we should just keep this logic as-is, and accept some disconnect in the updated DAOs? Or do you think that's dangerous? The Key Value DAO is well tested currently, so it should be pretty immune to regressions.

Reply:
As #24969 is now merged, this comment is no longer valid